Core/scripts/dirac-install.py

```diff
@@ -1021,296 +1053,313 @@ def fixBuildPaths():
+      logERROR("Requirements installation script %s failed. Check %s.err" % (reqScript,
+                                                                             reqScript))
   return True
 
 ####
 # End of helper functions
 ####
 
-cmdOpts = ( ( 'r:', 'release=', 'Release version to install' ),
-            ( 'l:', 'project=', 'Project to install' ),
-            ( 'e:', 'extensions=', 'Extensions to install (comma separated)' ),
-            ( 't:', 'installType=', 'Installation type (client/server)' ),
-            ( 'i:', 'pythonVersion=', 'Python version to compile (27/26)' ),
-            ( 'p:', 'platform=', 'Platform to install' ),
-            ( 'P:', 'installationPath=', 'Path where to install (default current working dir)' ),
-            ( 'b', 'build', 'Force local compilation' ),
-            ( 'g:', 'grid=', 'lcg tools package version' ),
-            ( '  ', 'no-lcg-bundle', 'lcg tools not to be installed' ),
-            ( 'B', 'noAutoBuild', 'Do not build if not available' ),
-            ( 'v', 'useVersionsDir', 'Use versions directory' ),
-            ( 'u:', 'baseURL=', "Use URL as the source for installation tarballs" ),
-            ( 'd', 'debug', 'Show debug messages' ),
-            ( 'V:', 'installation=', 'Installation from which to extract parameter values' ),
-            ( 'X', 'externalsOnly', 'Only install external binaries' ),
-            ( 'M:', 'defaultsURL=', 'Where to retrieve the global defaults from' ),
-            ( 'h', 'help', 'Show this help' ),
-            ( 'T:', 'Timeout=', 'Timeout for downloads (default = %s)' )
-          )
+
+cmdOpts = (('r:', 'release=', 'Release version to install'),
+           ('l:', 'project=', 'Project to install'),
+           ('e:', 'extensions=', 'Extensions to install (comma separated)'),
+           ('t:', 'installType=', 'Installation type (client/server)'),
+           ('i:', 'pythonVersion=', 'Python version to compile (27/26)'),
+           ('p:', 'platform=', 'Platform to install'),
+           ('P:', 'installationPath=', 'Path where to install (default current working dir)'),
+           ('b', 'build', 'Force local compilation'),
+           ('g:', 'grid=', 'lcg tools package version'),
+           ('  ', 'no-lcg-bundle', 'lcg tools not to be installed'),
+           ('B', 'noAutoBuild', 'Do not build if not available'),
+           ('v', 'useVersionsDir', 'Use versions directory'),
+           ('u:', 'baseURL=', "Use URL as the source for installation tarballs"),
+           ('d', 'debug', 'Show debug messages'),
+           ('V:', 'installation=', 'Installation from which to extract parameter values'),
+           ('X', 'externalsOnly', 'Only install external binaries'),
+           ('M:', 'defaultsURL=', 'Where to retrieve the global defaults from'),
+           ('h', 'help', 'Show this help'),
+           ('T:', 'Timeout=', 'Timeout for downloads (default = %s)'),
+           ('O:', 'diracos-version=', 'the version of the DIRAC OS'),
```

Comment:
Given that at the next line the switch is "dirac-os", I would prefer that this switch is "dirac-os-version". And I think we can safely drop the "O:" switch and live only with the extended one.

Reply:
right